### PR TITLE
Allow silencing version logging with a system property

### PR DESCRIPTION
**Pull request by gsmet**
/cc @dmlloyd @yrodiere

### CODE REVIEW

**Review by yrodiere — APPROVED** [review]
LGTM

Context: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Disable.20version.20logging

File: src/main/java/org/jboss/threads/Version.java [hunk 1]
```diff
@@ -48,9 +51,17 @@ private Version() {
         }
         JAR_NAME = jarName;
         VERSION_STRING = versionString;
+
+        boolean logVersion = true;
         try {
-            Messages.msg.version(versionString);
+            logVersion = !"false".equalsIgnoreCase(System.getProperty(JBOSS_LOG_VERSIONS));
         } catch (Throwable ignored) {}
```
Comment (gsmet) [inline src/main/java/org/jboss/threads/Version.java:58]: Decided to be extra defensive to keep the original behavior even if `System.getProperty` throws something.

Comment (darranl) [reply src/main/java/org/jboss/threads/Version.java:58]: +1 was just thinking what would happen if a security manager prevented the call.

**Review by jamezp — APPROVED** [review]

File: src/main/java/org/jboss/threads/Version.java [hunk 2]
```diff
@@ -31,6 +31,9 @@ public final class Version {
     private Version() {
     }
 
+    // No specific prefix here, this controls the behavior of multiple JBoss libraries
+    private static final String JBOSS_LOG_VERSIONS = "jboss.log-versions";
```
Comment (gsmet) [inline src/main/java/org/jboss/threads/Version.java:35]: Wondering if we should make this public for convenience but I'm not sure we want to commit to it always being around.

Comment (jamezp) [reply src/main/java/org/jboss/threads/Version.java:35]: This is more of a convention so I'd opt for it not being public. More than likely too you'd want the property configured before you access the type as logging would need to be configured first.

Comment (gsmet) [reply src/main/java/org/jboss/threads/Version.java:35]: Ah ah yeah. I thought of that yesterday and then forgot today why I didn’t post the proposal in the end :).

File: src/main/java/org/jboss/threads/Version.java [hunk 3]
```diff
@@ -48,9 +51,17 @@ private Version() {
         }
         JAR_NAME = jarName;
         VERSION_STRING = versionString;
+
+        boolean logVersion = true;
         try {
-            Messages.msg.version(versionString);
+            logVersion = !"false".equalsIgnoreCase(System.getProperty(JBOSS_LOG_VERSIONS));
```
Comment (dmlloyd) [inline src/main/java/org/jboss/threads/Version.java:57]: I would strongly prefer usage of `parseBoolean` here, and use `"true"` as the default value for the property get.

Comment (gsmet) [reply src/main/java/org/jboss/threads/Version.java:57]: @dmlloyd yes let’s go with your PR.

**Pull request closed without merging**